Improving Python 2 and 3 compatibility

larsonsb opened the pull request:
@hltbra I made some changes to make the code compatible with both Python 2 and 3, so that we can consider using this instead of milieu in pipeapp (which of course we are currently also working to make Python 2 and 3 compatible). The tests pass when running in either Python 2 and 3. Let me know any thoughts you have!

hltbra commented on requirements.txt:
```diff
@@ -1,4 +1,3 @@
-couleur==0.5.0
```
You should include six in the `requirements.txt`, it doesn't come by default with Python 2.

Unrelated to this PR, but I'd make the other requirements more relaxed instead of pinning specific versions (maybe try minimum versions, such as `mypackage>=1.2.3`.

hltbra commented on envelop/__init__.py:
```diff
@@ -17,9 +17,10 @@
 
 from __future__ import unicode_literals
+from six.moves.urllib.parse import urlparse
+from builtins import str as text
```
Python 2.x doesn't have the module `builtins`, we should probably have something from six (`six.text_type` maybe?) or define it ourselves.

larsonsb replied:
Makes sense. Another alternative could be to include `future>=0.16.0` in `requirements.txt`? `builtins` is installed through that.

hltbra replied:
Unless we use `futures` directly, I wouldn't do this because we would need to install a library we don't use and all of its dependencies (in the future they may change the lib to have extra dependencies). If we're already depending on `six` and it provides the type we want, I'd use it; otherwise define a small try/except for the type. I guess six implements `text_type` as

```python
try:
    from builtins import str as text_type
except ImportError:
    text_type = unicode
```

larsonsb commented:
@hltbra 
Added `six` to `requirements.txt`
Added `.travis.yml`
Used `six.text_type` to replace `unicode()`, instead of `builtins.str`
Also changed `commands` to `subprocess` (I had meant to include this commit when I last pushed; but inadvertently missed it)

hltbra commented on tests/functional/test_main.py:
```diff
@@ -1,7 +1,10 @@
-import commands
+from future import standard_library
```
The `future` package should be added to `requirements-dev.txt` if you want to use it in the tests.

larsonsb replied:
Gotcha. Thanks. Added it. I should probably get in the habit of putting packages in the requirements files as soon as I import them somewhere.
